Tools/autotest/rover.py

```diff
@@ -5347,6 +5349,84 @@ def test_slew_rate(self):
+            if now - tstart > 10:
+                raise AutoTestTimeoutException("Didn't get to speed")
+            self.mav.mav.set_attitude_target_send(
+                0, # time_boot_ms
+                target_sysid,
+                target_compid,
+                mavutil.mavlink.ATTITUDE_TARGET_TYPEMASK_BODY_ROLL_RATE_IGNORE |
+                mavutil.mavlink.ATTITUDE_TARGET_TYPEMASK_BODY_PITCH_RATE_IGNORE |
+                mavutil.mavlink.ATTITUDE_TARGET_TYPEMASK_ATTITUDE_IGNORE,
+                mavextra.euler_to_quat([0,
+                                        math.radians(0),
+                                        math.radians(0)]), # att
+                0, # yaw rate (rad/s)
+                0, # pitch rate
+                0, # yaw rate
+                1) # thrust
+
+            msg = self.mav.recv_match(type='VFR_HUD', blocking=True, timeout=1)
+            if msg is None:
+                raise NotAchievedException("No VFR_HUD message")
+            if msg.groundspeed > 5:
+                break
+        self.disarm_vehicle()
+
+    def SET_POSITION_TARGET_LOCAL_NED(self, target_sysid=None, target_compid=1):
+        if target_sysid is None:
+            target_sysid = self.sysid_thismav()
+        self.change_mode('GUIDED')
+        self.wait_ready_to_arm()
+        self.arm_vehicle()
+        tstart = self.get_sim_time()
+        while True:
+            now = self.get_sim_time_cached()
+            if now - tstart > 10:
+                raise AutoTestTimeoutException("Didn't get to speed")
+            self.mav.mav.set_position_target_local_ned_send(
+                0, # time_boot_ms
+                target_sysid,
+                target_compid,
+                mavutil.mavlink.MAV_FRAME_LOCAL_NED,
+                mavutil.mavlink.POSITION_TARGET_TYPEMASK_VX_IGNORE |
+                mavutil.mavlink.POSITION_TARGET_TYPEMASK_VY_IGNORE |
+                mavutil.mavlink.POSITION_TARGET_TYPEMASK_VZ_IGNORE |
+                mavutil.mavlink.POSITION_TARGET_TYPEMASK_AX_IGNORE |
+                mavutil.mavlink.POSITION_TARGET_TYPEMASK_AY_IGNORE |
+                mavutil.mavlink.POSITION_TARGET_TYPEMASK_AZ_IGNORE |
+                mavutil.mavlink.POSITION_TARGET_TYPEMASK_YAW_IGNORE |
+                mavutil.mavlink.POSITION_TARGET_TYPEMASK_YAW_RATE_IGNORE,
+                30.0,  # pos-x
+                30.0,  # pos-y
+                0,     # pos-z
+                0,     # vel-x
+                0,     # vel-y
+                0,     # vel-z
+                0,     # acc-x
+                0,     # acc-y
+                0,     # acc-z
+                0,     # yaw
+                0,     # yaw rate
+            )
+
+            msg = self.mav.recv_match(type='VFR_HUD', blocking=True, timeout=1)
+            if msg is None:
+                raise NotAchievedException("No VFR_HUD message")
+            self.progress("speed=%f" % msg.groundspeed)
+            if msg.groundspeed > 5:
```

Comment:
Hum this is a wrong test... You should do like I have done on global_ned and check that the right position is reached.
What we want to check here is the position controler not the speed one

Reply:
Yeah, in this case I only wanted to make sure we *acted* on the message, given the nature of the PR.

I *believe* you have a pending PR which adds all sorts of cool tests for this and didn't want to replicate it!

Reply:
ok then ! we can do better later. I don't have start working on those test yet but that is a matter of time